modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/service/impl/RedirectEntryLocalServiceImpl.java

```diff
@@ -407,6 +409,24 @@ private void _validate(String destinationURL, String sourceURL)
 		if (Validator.isNull(sourceURL)) {
 			throw new RequiredRedirectEntrySourceURLException();
 		}
+
+		_validateURL(sourceURL);
+	}
+
+	private void _validateURL(String sourceURL)
+		throws LayoutFriendlyURLException {
```

Comment:
I am not sure if we should use Layout exceptions here. We can discuss it tomorrow.